### PR TITLE
OTel Access Logger: support additional stat prefix

### PR DESCRIPTION
**Pull request by TAOXUY**
As described in the API, this allows to distinguish emitted statistics between configured open telemetry access loggers in the logger chain. If empty, the default `access_logs.open_telemetry_access_log.` will be used.

It is based on https://github.com/envoyproxy/envoy/pull/34072 so please review from the 4th commit.

### CODE REVIEW

File: api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto [hunk 1]
```diff
@@ -46,4 +46,9 @@ message OpenTelemetryAccessLogConfig {
   // See 'attributes' in the LogResource proto for more details.
   // Example: ``attributes { values { key: "user_agent" value { string_value: "%REQ(USER-AGENT)%" } } }``.
   opentelemetry.proto.common.v1.KeyValueList attributes = 3;
+
+// Optional additional prefix to use when emitting statistics. This allows to distinguish
+  // emitted statistics between configured open telemetry access loggers in the logger chain.
+  // If empty, the default `access_logs.open_telemetry_access_log.` will be used.
+  string stat_prefix = 6;
```
Comment (adisuissa) [inline api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto:53]: Please add a PGV constraint to ensure that if this is populated, it is not empty:
`[(validate.rules).string = {min_len: 1}]`

File: api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto [hunk 2]
```diff
@@ -46,4 +46,9 @@ message OpenTelemetryAccessLogConfig {
   // See 'attributes' in the LogResource proto for more details.
   // Example: ``attributes { values { key: "user_agent" value { string_value: "%REQ(USER-AGENT)%" } } }``.
   opentelemetry.proto.common.v1.KeyValueList attributes = 3;
+
+// Optional additional prefix to use when emitting statistics. This allows to distinguish
+  // emitted statistics between configured open telemetry access loggers in the logger chain.
+  // If empty, the default `access_logs.open_telemetry_access_log.` will be used.
```
Comment (adisuissa) [inline api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto:52]:
```suggestion
  // If not provided, the default `access_logs.open_telemetry_access_log.` prefix will be used.
```

File: api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto [hunk 3]
```diff
@@ -46,4 +46,9 @@ message OpenTelemetryAccessLogConfig {
   // See 'attributes' in the LogResource proto for more details.
   // Example: ``attributes { values { key: "user_agent" value { string_value: "%REQ(USER-AGENT)%" } } }``.
   opentelemetry.proto.common.v1.KeyValueList attributes = 3;
+
+// Optional additional prefix to use when emitting statistics. This allows to distinguish
```
Comment (adisuissa) [inline api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto:50]: Is this "additional prefix" or replacement prefix?
Also consider adding a reference to the statistics (IIUC it should be something similar to:
```
:ref:`statistics <config_access_log_stats>`
```

**Review by adisuissa — COMMENTED** [review]
Thanks for adding new features.
I've left API-related comments. This change will also require a release note.
Assigning to Yan as code-owner.
/assign @yanavlasov

**Comment by yanavlasov** [comment]
I will review once #34072 is submitted.
/wait

File: api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto (hunk 2 above)
Comment (TAOXUY) [reply api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto]: Updated.

File: api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto (hunk 3 above)
Comment (TAOXUY) [reply api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto]: PTAL. reworded.

File: api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto (hunk 1 above)
Comment (TAOXUY) [reply api/envoy/extensions/access_loggers/open_telemetry/v3/logs_service.proto:53]: Unnecessary for optional string field?

**Review by adisuissa — COMMENTED** [review]
/lgtm api

**Review by adisuissa — COMMENTED** [review]
Please add a release note.

**Comment by TAOXUY** [comment]
> Please add a release note.

DONE

**Review by adisuissa — COMMENTED** [review]
/lgtm api

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**